testbed/core/models.py

```diff
@@ -1,38 +1,64 @@
 import logging
 from django.db import models
 from django.contrib.auth.models import User
+from django.contrib.auth import get_user_model
 from django.core.exceptions import ValidationError
 from datetime import timezone
 
 logger = logging.getLogger(__name__)
 
+class ActorManager(models.Manager):
+    # Create both source and destination actors for a user
+    def create_actors_for_user(self, user):
+        source = self.create(
+            user=user,
+            role=Actor.ROLE_SOURCE,
+        )
+        destination = self.create(
+            user=user,
+            role=Actor.ROLE_DESTINATION,
+        )
 
-def validate_username(value):
-    if " " in value or not value.isalnum():
-        raise ValidationError("Username must be alphanumeric and contain no spaces.")
-
-    if len(value) < 3:
-        raise ValidationError("Username must be at least 3 characters.")
+        return source, destination
 
 
 class Actor(models.Model):
-    user = models.OneToOneField(User, on_delete=models.CASCADE, related_name="actor")
-    username = models.CharField(
-        max_length=100, unique=True, validators=[validate_username]
-    )
-    full_name = models.CharField(max_length=100)
+    ROLE_SOURCE = "source"
+    ROLE_DESTINATION = "destination"
+    ROLE_CHOICES = [
+        (ROLE_SOURCE, "Source Service"),
+        (ROLE_DESTINATION, "Destination Service"),
+    ]
+
+    user = models.ForeignKey(get_user_model(), on_delete=models.CASCADE, related_name="actors")
+    role = models.CharField(max_length=20, choices=ROLE_CHOICES)
     created_at = models.DateTimeField(auto_now_add=True)
     updated_at = models.DateTimeField(auto_now=True)
     previously = models.JSONField(default=list, null=True, blank=True)
 
+    objects = ActorManager()
+
+    class Meta:
+        constraints = [
+            models.UniqueConstraint(
```

Comment:
I don't think we need a constraint here?  at least not on the SQL level - if we wanted to add a feature to have more than one kind of Actor to pull from we'd need to then remove this constraint - and it's not adding anything I can think of